rules/rbe_repo.bzl

```diff
@@ -63,65 +103,154 @@ Add to your WORKSPACE file the following:
     digest = "sha256:deadbeef",
   )
 
-For values of <latest_release> and other placeholders above, please see
-the WORKSPACE file in this repo.
 
-This rule depends on the value of the environment variable "RBEAUTOCONF_ROOT"
-when output_base is used.
-This env var should be set to point to the absolute path root of your project.
-Use the full absolute path to the project root (i.e., no '~', '../', or
-other special chars).
+For use case 3.2. If you are using a custom container, and don't
+want to generate configs every time. The setup is as follows:
 
-There are two modes of using this repo rules:
-  1 - When output_base set (recommended if using a custom toolchain container
-    image; env var "RBEAUTOCONF_ROOT" is required), running the repo rule
-    target will copy the toolchain config files to the output_base folder in
-    the project sources.
-    After that, you can run an RBE build pointing your crosstool_top flag to the
-    produced files. If output_base is set to "rbe-configs" (recommended):
+Define a repo (can be the same where you host the sources you want
+to build on RBE or a separate one) which will host your published
+toolchain configs. We call this the "toolchain-config-repo" in the
+below
 
-      bazel build ... \
-                --crosstool_top=//rbe-configs/bazel_{bazel_version}/cc:toolchain \
-                --host_javabase=//rbe-configs/bazel_{bazel_version}/java:jdk \
-                --javabase=//rbe-configs/bazel_{bazel_version}/java:jdk \
-                --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8 \
-                --java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8 \
-                --extra_execution_platforms=//rbe-configs/bazel_{bazel_version}/config:platform \
-                --host_platform=//rbe-configs/bazel_{bazel_version}/config:platform \
-                --platforms=//rbe-configs/bazel_{bazel_version}/config:platform \
-                --extra_toolchains=//rbe-configs/bazel_{bazel_version}/config:cc-toolchain \
-                ... <other rbe flags> <build targets>
-
-    We recommend you check in the code in //rbe-configs/bazel_{bazel_version}
-    so that most devs/your CI typically do not need to run this repo rule
-    in order to do a remote build (i.e., once files are checked in,
-    you do not need to run this rule until there is a new version of Bazel
-    you want to support running with, or you need to update your container).
-
-  2 - When output_base is not set (recommended for users of the
-    rbe-ubuntu 16_04 images - env var "RBEAUTOCONF_ROOT" is not required),
-    running this rule will create targets in the
-    external repository (e.g., rbe_default) which can be used to point your
-    flags to:
+Define a toolchain_config_suite_spec, which specifies a container
+(repo + registry) and an output_base, the relative path, within
+the toolchain-config-repo where toolchain configs will be published to.
```

Comment:
Add a note that users need to commit the configs separately.

Reply:
added in line 197